LaTeX/LaTeX.sublime-syntax

```diff
@@ -4,554 +4,907 @@
+        - include: main
+    - match: '((\\)begin)(\{)\s*(description\*?)\s*(\})'
       captures:
         1: support.function.be.latex
-        2: punctuation.definition.function.latex
-        3: punctuation.definition.arguments.begin.latex
+        2: punctuation.definition.backslash.latex
+        3: punctuation.definition.brace.begin.latex
         4: variable.parameter.function.latex
-        5: punctuation.definition.arguments.end.latex
-      push:
-        - meta_scope: meta.function.environment.math.latex
-        - meta_content_scope: string.other.math.block.environment.latex
-        - match: |-
-            (?x)
-            	(?:\s*)										# Optional whitespace
-            	((\\)end)									# Marker - Function
-            	(\{)										# Open Bracket
-            		(\4)				# Previous capture from begin
-            	(\})										# Close Bracket
-            	(?:\s*\n)?				# Match to end of line absent of content
+        5: punctuation.definition.brace.end.latex
+      push:
+        - meta_scope: meta.environment.list.description.latex
+        - match: '((\\)end)(\{)\s*(\4)\s*(\})'
           captures:
             1: support.function.be.latex
-            2: punctuation.definition.function.latex
-            3: punctuation.definition.arguments.begin.latex
+            2: punctuation.definition.backslash.latex
+            3: punctuation.definition.brace.begin.latex
             4: variable.parameter.function.latex
-            5: punctuation.definition.arguments.end.latex
+            5: punctuation.definition.brace.end.latex
           pop: true
-        - include: $top_level_main
-    - match: |-
-        (?x)
-        	(?:\s*)										# Optional whitespace
-        	((\\)begin)									# Marker - Function
-        	(\{)										# Open Bracket
-        		(array|tabular[xy*]?)
-        	(\})										# Close Bracket
-        	(\s*\n)?				# Match to end of line absent of content
+        - include: main
+
+  math-braces:
+    - match: '\{'
+      scope: punctuation.definition.brace.begin.latex
+      push:
+        - meta_scope: meta.group.braces.latex
+        - match: '\}'
+          scope: punctuation.definition.brace.end.latex
+          pop: true
+        - include: math-content
+
+  math-content:
+    # unique to latex
+    - include: verb
+    - include: text-decorators
+    - include: references
+    - include: begin-end-commands
+    # extended from tex
+    - include: scope:text.tex#greeks
+    - include: scope:text.tex#math-brackets
+    - include: math-braces
+    - include: boxes
+    - include: scope:text.tex#math-commands
+    - include: scope:text.tex#math-operators
+    - include: scope:text.tex#math-characters
+    - include: scope:text.tex#math-numerics
+    - include: general-constants
+
+  inline-math:
+    - match: \$
+      scope: string.other.math.latex punctuation.definition.string.begin.latex
+      push:
+        - meta_scope: meta.environment.math.latex
```

Comment:
I think its better, but we should also differ between `\(...\)` and `$...$`. The scopename could be similar to the convention of comments, i.e. `meta.environment.math.inline.dollar.latex`, `meta.environment.math.inline.paren.latex`, `meta.environment.math.block.brace.latex`, or something with `tex.latex` and `latex.latex` (for the commands, which come from tex and latex).